src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -143,6 +143,29 @@ impl ScalarFunc {
         Ok(Some(s))
     }
 
+    pub fn right<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
+        let i = try_opt!(self.children[1].eval_int(ctx, row));
+        if i <= 0 {
+            return Ok(Some(Cow::Owned(b"".to_vec())));
+        }
+        let len = s.chars().count();
+        if len > i as usize {
+            let t = s.chars();
+            return Ok(Some(Cow::Owned(
+                t.skip(len - i as usize)
```

Comment:
According to a little bench, re-alloc a string is much more expensive than use the in-place one:
```rust
#![feature(test)]
extern crate test;

#[cfg(test)]
mod tests {
    use test::Bencher;
    const S1: &'static str = "努曼格莫多夫整个UFC生涯都表现得十分冷静沉着，但在昨天的比赛之后，他让我们看到了他的另一面，而这另一面造成了非常不好的影响。";
    const S2: &'static str = "努曼格莫多夫";

    #[bench]
    fn bench_chars_skip(b: &mut Bencher) {
        let length = S2.chars().count();
        let pos = length / 2;
        b.iter(|| {
            let ss: String = S2.chars().skip(pos).collect();
            drop(ss);
        });
    }

    #[bench]
    fn bench_chars_next(b: &mut Bencher) {
        let length = S2.chars().count();
        let pos = length / 2;
        b.iter(|| {
            let mut chars = S2.chars();
            for _ in 0..pos {
                chars.next();
            }
            let ss: &'static str = chars.as_str();
            drop(ss);
        })
    }
}
```
> test tests::bench_chars_next ... bench:          11 ns/iter (+/- 1)
> test tests::bench_chars_skip ... bench:         207 ns/iter (+/- 0)
So could you change the implement?